apps/frontend/src/app/5_pages/MarketMakingPage/components/AdjustAndDepositModal/AdjustAndDepositModal.tsx

```diff
@@ -52,35 +64,103 @@ export const AdjustAndDepositModal: FC<AdjustAndDepositModalProps> = ({
 }) => {
   const [adjustType, setAdjustType] = useState(AdjustType.Deposit);
   const [value, setValue, amount] = useWeiAmountInput('');
+  const { account } = useAccount();
+  const { onDeposit, onWithdraw } = useHandleMarketMaking(onClose);
+  const { balanceA, loadingA } = useGetUserInfo(pool);
+
+  const decimalAmount = useMemo(
+    (): Decimal => decimalic(amount.toString()),
+    [amount],
+  );
+
+  const isAmountZero = useMemo(() => amount.isZero(), [amount]);
+  const { tokenPoolBalance } = useGetPoolBalance(
+    isAmountZero,
+    adjustType,
+    loadingA,
+    pool,
+    account,
+  );
+
+  const poolWeiAmount = calculatePoolWeiAmount(
+    Decimal.fromBigNumberString(amount.toString()),
```

Comment:
```suggestion
    decimalAmount,
```

Reply:
it will not work with `decimalAmount`, for more info you can check the values